diff --git a/src/components/CategoryItem.tsx b/src/components/CategoryItem.tsx
index cfef1317181c989271eddb7a4982f3a59ff92967..dfd0c29590248f5d3575b637165f7c4f0a46bd58 100644
--- a/src/components/CategoryItem.tsx
+++ b/src/components/CategoryItem.tsx
@@ -1,8 +1,9 @@
 import * as React from 'react'
-import { View, StyleSheet, TouchableOpacity } from 'react-native'
+import { View, StyleSheet } from 'react-native'
 import { KeyboardContext } from '../contexts/KeyboardContext'
 import type { CategoryNavigationItem } from '../types'
 import { Icon } from './Icon'
+import { Pressable } from "react-native-gesture-handler"
 
 type CategoryItemProps = {
   item: CategoryNavigationItem
@@ -19,7 +20,7 @@ export const CategoryItem = ({ item, index, handleScrollToCategory }: CategoryIt
   }
 
   return (
-    <TouchableOpacity onPress={handleSelect}>
+    <Pressable onPress={handleSelect} style={({ pressed }) => ({ opacity: pressed ? 0.5 : 1 })}>
       <View style={styles.container}>
         <Icon
           iconName={item.icon}
@@ -28,7 +29,7 @@ export const CategoryItem = ({ item, index, handleScrollToCategory }: CategoryIt
           activeColor={theme.category.iconActive}
         />
       </View>
-    </TouchableOpacity>
+    </Pressable>
   )
 }
 
diff --git a/src/components/EmojiStaticKeyboard.tsx b/src/components/EmojiStaticKeyboard.tsx
index ff00d44a76820dd32211264dbb9d1846d275a87f..6e2a36e9a8235abf354e84e89f102d4c40d64661 100644
--- a/src/components/EmojiStaticKeyboard.tsx
+++ b/src/components/EmojiStaticKeyboard.tsx
@@ -4,7 +4,6 @@ import {
   View,
   FlatList,
   Animated,
-  SafeAreaView,
   Platform,
   type NativeScrollEvent,
   type NativeSyntheticEvent,
@@ -17,6 +16,7 @@ import { SearchBar } from './SearchBar'
 import { useKeyboardStore } from '../store/useKeyboardStore'
 import { ConditionalContainer } from './ConditionalContainer'
 import { SkinTones } from './SkinTones'
+import { SafeAreaView } from 'react-native-safe-area-context';
 
 const isAndroid = Platform.OS === 'android'
 
diff --git a/src/components/SingleEmoji.tsx b/src/components/SingleEmoji.tsx
index c878fc2888642edbc2bb07906a8c53f4775c1d14..efc496d5c87670758d1e213fbc2a2f844ec96c62 100644
--- a/src/components/SingleEmoji.tsx
+++ b/src/components/SingleEmoji.tsx
@@ -2,13 +2,12 @@ import * as React from 'react'
 import {
   View,
   Text,
-  TouchableOpacity,
   StyleSheet,
-  type GestureResponderEvent,
   type ViewStyle,
   type StyleProp,
 } from 'react-native'
 import type { EmojiSizes, JsonEmoji } from '../types'
+import { Pressable } from "react-native-gesture-handler"
 
 type Props = {
   item: JsonEmoji
@@ -23,7 +22,7 @@ type Props = {
 export const SingleEmoji = React.memo(
   (p: Props) => {
     const handlePress = () => p.onPress(p.item)
-    const handleLongPress = (e: GestureResponderEvent) => {
+    const handleLongPress = (e: any) => {
       // @ts-ignore
       e.target.measure((_x, _y, width, height) => {
         p.onLongPress(p.item, p.index, { width, height })
@@ -31,15 +30,15 @@ export const SingleEmoji = React.memo(
     }
 
     return (
-      <TouchableOpacity
+      <Pressable
         onPress={handlePress}
         onLongPress={handleLongPress}
-        style={styles.container}
+        style={({ pressed }) => [styles.container, { opacity: pressed ? 0.5 : 1 }]}
       >
         <View pointerEvents={'none'} style={[styles.emojiWrapper, p.selectedEmojiStyle]}>
           <Text style={[styles.emoji, { fontSize: p.emojiSize }]}>{p.item.emoji}</Text>
         </View>
-      </TouchableOpacity>
+      </Pressable>
     )
   },
   (prevProps, nextProps) => prevProps.isSelected === nextProps.isSelected,
